ql/src/java/org/apache/hadoop/hive/ql/exec/persistence/MapJoinKey.java

```diff
@@ -85,6 +84,7 @@ public static MapJoinKey read(Output output, MapJoinObjectSerDeContext context,
     SUPPORTED_PRIMITIVES.add(PrimitiveCategory.BINARY);
     SUPPORTED_PRIMITIVES.add(PrimitiveCategory.VARCHAR);
     SUPPORTED_PRIMITIVES.add(PrimitiveCategory.CHAR);
+    SUPPORTED_PRIMITIVES.add(PrimitiveCategory.DECIMAL);
```

Comment:
@mustafaiman , do we need to include the check on the scale/precision for decimal type?

Reply:
Check is not necessary. During planning, common type is used for join keys. This is true when `hive.cbo.enable` is `true` or `false`. See `FunctionRegistry.getCommonClassForComparison`.